modules/graph_ql/main.tf

```diff
@@ -1,4 +1,133 @@
-data "aws_iam_policy_document" "graphql_task_execution_policy_document" {
+# Local Variables
+locals {
+  graphql_port = 5000
+}
+
+resource "random_id" "lb_name" {
+  byte_length = 4
+}
+
+resource "aws_security_group" "gql_security_group" {
+  name        = "${var.prefix}-gql"
+  description = "Allow inbound communication on container port."
+  vpc_id      = var.vpc_id
+
+  ingress {
+    description      = "Container port communication."
+    from_port        = local.graphql_port
+    to_port          = local.graphql_port
+    protocol         = "tcp"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = var.tags
+}
+
+# Application load balancer
+resource "aws_lb" "gql_app_lb" {
```

Comment:
we can also enable access logs to track activities if desired.

Reply:
Gave it a shot, but looks like it requires additional permissions on the `internal` system bucket. Given the difficulties we had the last time we suggested s3 policy updates, recommend we hold off on this until the system is ready for customer integration/migration.

Reply:
https://bugs.earthdata.nasa.gov/browse/ORCA-583